extensions/data-transfer/portability-data-transfer-koofr/src/main/java/org/datatransferproject/transfer/koofr/photos/KoofrPhotosImporter.java

```diff
@@ -74,13 +75,17 @@ public ImportResult importItem(
       PhotosContainerResource resource)
       throws Exception {
     KoofrClient koofrClient = koofrClientFactory.create(authData);
-
     monitor.debug(
         () ->
             String.format(
                 "%s: Importing %s albums and %s photos before transmogrification",
                 jobId, resource.getAlbums().size(), resource.getPhotos().size()));
 
+   TimeZone userTimeZone = jobStore.findJob(jobId).userTimeZone();
+    if (null != userTimeZone) {
+      titleDateFormat.setTimeZone(userTimeZone);
```

Comment:
If `KoofrPhotosImporter` is used in a context where multiple jobs can be processed with the same instance concurrently this will cause a data race between multiple jobs.

`titleDateFormat` should be cloned for each `importItem` call and `setTimeZone` can be called for the cloned object.

Reply:
+1 you can probably keep a cache from job id --> timezone to avoid looking up the job each time import is called but it should be separate per job.

Reply:
Thank you for your suggestions! I will try to fix it.